src/TLuaInterpreter.cpp

```diff
@@ -17158,49 +17157,90 @@ void TLuaInterpreter::initIndenterGlobals()
+    // other OSes and it keeps things simpler if they all use it:
+    // clang-format off
+    luaL_dostring(pIndenterState.get(), R"LUA(function toNativeSeparators(rawPath)
+  if package.config:sub(1,1) == '\\' then
+    return string.gsub(rawPath, '/', '\\')
+  end
 
-    // if using LuaJIT, adjust the cpath to look in /usr/lib as well - it doesn't by default
-    luaL_dostring(pIndenterState.get(), "if jit then package.cpath = package.cpath .. ';/usr/lib/lua/5.1/?.so;/usr/lib/x86_64-linux-gnu/lua/5.1/?.so' end");
+  assert((package.config:sub(1,1) == '/'), "package path directory separator is neither '\\' nor '/' and cannot be handled")
 
-    //AppInstaller on Linux would like the search path to also be set to the current binary directory
-    luaL_dostring(pIndenterState.get(), QStringLiteral("package.cpath = package.cpath .. ';%1/lib/?.so'")
-                  .arg(QCoreApplication::applicationDirPath())
-                  .toUtf8().constData());
-#elif defined(Q_OS_WIN32)
-    // For Qt Creator builds, add search paths one and two levels up from here, then a 3rdparty directory:
-    luaL_dostring(pIndenterState.get(),
-                  QStringLiteral("package.path = [[%1\\?.lua;%2\\..\\3rdparty\\?.lua;%2\\..\\..\\3rdparty\\?.lua;]] .. package.path")
-                          .arg(QByteArray(LUA_DEFAULT_PATH), QDir::toNativeSeparators(QCoreApplication::applicationDirPath()))
-                          .toUtf8().constData());
+  return string.gsub(rawPath, '\\', '/')
+end)LUA");
+    // clang-format on
+
+    // 1 installed *nix case - probably not applicable to Windows
+    //     "LUA_DEFAULT_PATH/?.lua" (if defined and not empty)
+    if (!QStringLiteral(LUA_DEFAULT_PATH).isEmpty()) {
+        additionalLuaPaths << QStringLiteral(LUA_DEFAULT_PATH "/?.lua");
+    }
+    // 2 AppImage (directory of executable) - not needed for Wndows:
+    //     "<applicationDirectory>/?.lua"
+#if ! defined (Q_OS_WIN32)
+    additionalLuaPaths << QStringLiteral("%1/?.lua").arg(appPath);
 #endif
+    // 3 QMake shadow builds without CONFIG containing "debug_and_release" but
+    //    with "debug_and_release_target" (default on most OS but NOT Windows):
+    //     "<applicationDirectory>/../3rdparty/?.lua"
+    additionalLuaPaths << QStringLiteral("%1/../3rdparty/?.lua").arg(appPath);
+    // 4 QMake shadow builds with CONFIG containing "debug_and_release" AND
+    //   "debug_and_release_target" (usually Windows):
+    //     "<applicationDirectory>/../../3rdparty/?.lua"
+    additionalLuaPaths << QStringLiteral("%1/../../3rdparty/?.lua").arg(appPath);
+    // 5 CMake shadow builds
+    //    "<applicationDirectory>/../../mudlet/3rdparty/?.lua"
+    // NOTE: In the cmake case the shadow build directory is alongside the base
+    // directory of the extracted source package so the source code MUST be
+    // extracted into a known directory name which is assumed here to be
+    // "mudlet"!
```

Comment:
Having this information here is the equivalent of hiding it really, really well inside 17,000 lines! This isn't the right place for it: at most, it should be added to the compiling instructions for cmake.

Reply:
Okay - I guess... :neutral_face: